pulp_smash/tests/docker/cli/test_sync_publish.py

```diff
@@ -0,0 +1,281 @@
+            upstream_name=_UPSTREAM_NAME,
+        )
+        cls.completed_proc = docker_utils.repo_sync(cls.cfg, cls.repo_id)
+
+        cls.client = api.Client(cls.cfg)
+
+    def test_app_file_registry_id(self):
+        """Assert that the v2 app file has the correct repo-registry-id."""
+        app_file = self._get_app_file()
+        self.assertEqual(app_file['repo-registry-id'], self.repo_id)
+
+    def test_app_file_repository(self):
+        """Assert that the v2 app file has the correct repository."""
+        app_file = self._get_app_file()
+        self.assertEqual(app_file['repository'], self.repo_id)
+
+    def test_app_file_url(self):
+        """Assert that the v2 app file has the correct url."""
+        app_file = self._get_app_file()
+        cmd = 'hostname --fqdn'
+        fqdn = cli.Client(self.cfg).run(cmd.split()).stdout.strip()
+        self.assertEqual(app_file['url'],
+                         'https://{}/pulp/docker/v2/{}/'.format(
+                             fqdn, self.repo_id))
+
+    def test_app_file_version(self):
+        """Assert that the v2 app file has the correct version."""
+        app_file = self._get_app_file()
+        self.assertEqual(app_file['version'], 2)
+
+    def test_app_file_protected(self):
+        """The repo should not be marked as protected."""
+        app_file = self._get_app_file()
+        self.assertEqual(app_file['protected'], False)
+
+    def test_app_file_type(self):
+        """Assert that the v2 app file has the correct type."""
+        app_file = self._get_app_file()
+        self.assertEqual(app_file['type'], 'pulp-docker-redirect')
+
+    def test_blob_digests(self):
+        """Assert that we can retrieve all of the Blobs correctly."""
+        manifest = self._get_manifest()
+        # A list of 2-tuples of (expected_digest, actual_digest)
+        digests = []
+
+        for blob_sum in manifest['fsLayers']:
+            blob_sum = blob_sum['blobSum']
+            blob = self.client.get(
+                '/pulp/docker/v2/{}/blobs/{}'.format(
+                    self.repo_id, blob_sum)).content
+            algo, expected_digest = blob_sum.split(':')
+            hasher = getattr(hashlib, algo)()
+            hasher.update(blob)
+            digests.append((expected_digest, hasher.hexdigest()))
+
+        self.assertTrue(all([t[0] == t[1] for t in digests]))
+
+    def test_manifest_fslayers(self):
+        """
+        Assert that the manifest fsLayers is a list of dictionaries.
+
+        Each dictionary should have one key (blobSum) which should index a
+        string.
+        """
+        manifest = self._get_manifest()
+        self.assertTrue(isinstance(manifest['fsLayers'], list))
+        self.assertTrue(
+            all([isinstance(l, dict) for l in manifest['fsLayers']]))
+        self.assertTrue(
+            all([l.keys() == ['blobSum'] for l in manifest['fsLayers']]))
+        self.assertTrue(
+            all([isinstance(l['blobSum'], (str, unicode))
```

Comment:
Yeah I noticed that on Friday but it was 5 PM so I figured I'd save it for Monday. The type solution seems good. I was considering using the six library to fix this. Do you prefer your technique? It does avoid adding a dependency so that's good.